.github/workflows/scanning.yml

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2020, 2021 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Scan repository and containers for security vulnerabilities, secrets and
+# misconfigurations.
+
+name: Vulnerability and Misconfiguration Scan
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  scan:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Maven
+        uses: stCarolas/setup-maven@v4.3
+        with:
+          maven-version: 3.8.6
+
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: "temurin"
+          java-version: "17"
+          cache: "maven"
+
+      - name: Build all Docker images
+        run: |
+          mvn clean package -B -e -DCI=$CI -DskipTests -Pbuild-docker-image
```

Comment:
we should also activate the `metrics-prometheus` profile because it has an impact on the dependencies being added